modules/gitops-argo-cd-round-robin-sharding-algorithm.adoc

```diff
@@ -0,0 +1,235 @@
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Configure the sharding algorithm to `round-robin`:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"env":[{"name":"ARGOCD_CONTROLLER_SHARDING_ALGORITHM","value":"round-robin"}]}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Verify that the number of Argo CD Application Controller pods corresponds with the number of set replicas.
++
+[source,terminal]
+----
+$ oc get pods -l app.kubernetes.io/name=<argocd_instance>-application-controller -n <namespace>
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                        READY   STATUS    RESTARTS   AGE
+<argocd_instance>-application-controller-0   1/1     Running   0          11s
+<argocd_instance>-application-controller-1   1/1     Running   0          32s
+<argocd_instance>-application-controller-2   1/1     Running   0          22s
+----
+
+. Verify that the sharding is enabled with `round-robin` as the sharding algorithm:
++
+[source,terminal]
+----
+$ oc logs <argocd_application_controller_pod> -n <namespace>
+----
++
+.Example output snippet
+[source,terminal]
+----
+time="2023-12-13T09:05:34Z" level=info msg="ArgoCD Application Controller is starting" built="2023-12-01T19:21:49Z" commit=a3vd5c3df52943a6fff6c0rg181fth3248976299 namespace=<namespace> version=v2.9.2+c5ea5c4
+time="2023-12-13T09:05:34Z" level=info msg="Processing clusters from shard 1"
+time="2023-12-13T09:05:34Z" level=info msg="Using filter function:  round-robin" <1>
+time="2023-12-13T09:05:34Z" level=info msg="Using filter function:  round-robin"
+time="2023-12-13T09:05:34Z" level=info msg="appResyncPeriod=3m0s, appHardResyncPeriod=0s"
+----
+<1> Look for the `"Using filter function:  round-robin"` message.
+
+. Verify that the cluster distribution across shards is even:
+
+.. Set the log level to `debug`:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"logLevel":"debug"}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+.. View the logs and search for `processed by shard` to observe to which shard each cluster is attached:
++
+[source,terminal]
+----
+$ oc logs <argocd_application_controller_pod> -n <namespace> | grep "processed by shard"
+----
++
+.Example output snippet
+[source,terminal]
+----
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id= will be processed by shard 0" 
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id=068d8b26-6rhi-4w23-jrf6-wjjfyw833n23 will be processed by shard 1"
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id=836d8b53-96k4-f68r-8wq0-sh72j22kl90w will be processed by shard 2" 
+----
++
+In this example, 3 clusters are attached consecutively to shard 0, shard 1, and shard 2.
```

Comment:
In the console section example, you have this info as a callout, and here it's as body text after the example. 

Since the snippet example needs an introduction, you could say something like "The following example shows that 3 clusters are attached..." and move that sentence before the example. But I would do this for both examples for consistency.